code/modules/food_and_drinks/food/snacks_pie.dm

```diff
@@ -32,9 +32,13 @@
 /obj/item/reagent_containers/food/snacks/pie/cream/throw_impact(atom/hit_atom, datum/thrownthing/throwingdatum)
 	. = ..()
 	if(!.) //if we're not being caught
-		splat(hit_atom)
+		if(ishuman(throwingdatum.thrower))
+			var/mob/living/carbon/human/H = throwingdatum.thrower
+			splat(hit_atom,H)
+		else
+			splat(hit_atom)
```

Comment:
because it's not a keyword argument this should really be splat(hit_atom, null)

or the proc definition for splat should be
`/obj/item/reagent_containers/food/snacks/pie/cream/proc/splat(atom/movable/hit_atom, mob/living/carbon/human/H = null)`